error: Decompressor is not installed for grpc-encoding "gzip"

For compressed gRPC messages, I get the following error in grpc-dump (0.1.2, macOS):

```
"error":{"code":12,"message":"grpc: Decompressor is not installed for grpc-encoding \"gzip\""}
```

The request seems not to be forwarded to the server.

I installed grpc-dump via homebrew. Is there anything I can do to use compression?

Thanks for the effort. Unfortunately, my Go skills are even worse than I thought. I do not know how to build binaries from the branch. I tried

```sh
env GO111MODULE=on go get github.com/bradleyjkemp/grpc-tools@gzip-compression
```

, but did not find binaries anywhere.

And

```sh
go run grpc-dump/main.go 
```

yields an error:

```
# command-line-arguments
grpc-dump/main.go:52:30: undefined: dumpInterceptor
```

Sorry for being such a Go noob...

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,12 +2,13 @@
 
 ## v0.1.3 (unreleased)
 * Added proxy loop detection so that misconfiguration (e.g. missing/incorrent `--destination` flag) do not cause infinite loops and connection exhaustion.
+* `grpc-proxy` now supports requests with gzip compression (however requests are still proxied uncompressed).
 
 ## [v0.1.2](https://github.com/bradleyjkemp/grpc-tools/releases/tag/v0.1.2)
-* Fixed bug where the `--destination` flag didn't work (issue [#13](https://github.com/bradleyjkemp/grpc-tools/issues/13))
+* Fixed bug where the `--destination` flag didn't work (issue [#13](https://github.com/bradleyjkemp/grpc-tools/issues/13)).
 
 ## [v0.1.1](https://github.com/bradleyjkemp/grpc-tools/releases/tag/v0.1.1)
-* Added automatic detection of mkcert certificates in the current directory
+* Added automatic detection of mkcert certificates in the current directory.
 
 ## [v0.1.0](https://github.com/bradleyjkemp/grpc-tools/releases/tag/v0.1.0)
-* Added grpc-dump, grpc-fixture, grpc-replay
+* Added grpc-dump, grpc-fixture, grpc-replay.

--- a/grpc-proxy/proxy.go
+++ b/grpc-proxy/proxy.go
@@ -11,6 +11,7 @@ import (
 	"github.com/improbable-eng/grpc-web/go/grpcweb"
 	"github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
+	_ "google.golang.org/grpc/encoding/gzip"
 	"net"
 )
 

--- a/integration_test/test.sh
+++ b/integration_test/test.sh
@@ -59,7 +59,7 @@ ALL_PROXY=localhost:16354 curl -X POST 'https://grpc-web.github.io:1234/grpc.gat
     -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: en-US,en;q=0.9' -H 'custom-header-1: value1' \
     -H 'User-Agent: Mozilla/5.0' -H 'Content-Type: application/grpc-web+proto' -H 'Accept: */*' \
     -H 'X-Grpc-Web: 1' -H 'Cache-Control: no-cache' -H 'Referer: http://localhost:8081/echotest.html' \
-    -H 'Connection: keep-alive'
+    -H 'Connection: keep-alive' -H 'grpc-encoding: gzip'
 
 kill ${fixturePID}
 kill ${dumpPID}

